libs/openFrameworks/graphics/ofPixels.cpp

```diff
@@ -393,10 +393,10 @@ template<typename PixelType>
 void ofPixels_<PixelType>::setFromExternalPixels(PixelType * newPixels, size_t w, size_t h, ofPixelFormat _pixelFormat){
 	clear();
 	pixelFormat = _pixelFormat;
-	width= w;
+	width = w;
 	height = h;
 
-	pixelsSize = bytesFromPixelFormat(w,h,_pixelFormat);
+	pixelsSize = w * h;
```

Comment:
pixels size used to be: 
`	pixelsSize = bytesFromPixelFormat(w,h,_pixelFormat) / sizeof(PixelType);`

bytesFromPixelFormat is effectively: w * h * 24 / 8 ( for RGB ) 
so: newSize = w * h * 3;  for RGB

```
size_t ofPixels_<PixelType>::bytesFromPixelFormat(size_t w, size_t h, ofPixelFormat format){
	return w * h * pixelBitsFromPixelFormat(format) / 8;  <-- pixelBitsFromPixelFormat for rgb is 24 
}
```
a simpler fix would be: 

`pixelsSize = w*h * getNumChannels()`  

But maybe safer to first revert the commit where the regression was introduced and then do a PR with the bit depth support after?

Reply:
Part of the problem (historical ofPixels problem) is we have two different units
one is PixelType (it can be char, float, etc)

and the other are formats that doesn't have an uniform number of bytes per channel like RGB565 (5 + 6 + 5 total 16 bits) or NV12 (12 bits per channel)

but the array that stores pixel information will be always a PixelType (so we consider an "uncompressed" pixel type, to be worked on shaders).

we can go one way or another. 

if we support all the images format I suggest changing the Pixels array to uint8_t instead of PixelType and allocate everything following "bytesPerPixel"

Or keep everything as it is, and comment out all image types that doesn't follow PixelType bit depth.
makes sense?